File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/DTO/DTOProduct.java
package abbruzzese.webpage.webpage.server.webpage.DTO;

import java.time.LocalDate;
import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class DTOProduct {

    private String Nombre;

    private String Marca;

    private String Categoria;

    private String imageproduct;

    private LocalDate fecha;

    private String enlace;
    

    public DTOProduct( String Nombre,String Marca,String Descripcion,String Categoria,String imageproduct,LocalDate fecha, String enlace){

        this.Nombre=Nombre;
        this.Marca=Marca;
        this.Categoria=Categoria;
        this.imageproduct=imageproduct;
        this.fecha=fecha;
        this.enlace=enlace;
    }

    public DTOProduct(LocalDate fecha,String imageproduct, String enlace){

        this.fecha=fecha;
        this.imageproduct=imageproduct;
        this.enlace=enlace;
        
    }

    public DTOProduct(){}
    
    public String getNombre() {
        return Nombre;
    }

    public void setNombre(String nombre) {
        Nombre = nombre;
    }

    public String getMarca() {
        return Marca;
    }

    public void setMarca(String marca) {
        Marca = marca;
    }

    public String getCategoria() {
        return Categoria;
    }

    public void setCategoria(String categoria) {
        Categoria = categoria;
    }

    public String getImageproduct() {
        return imageproduct;
    }

    public void setImageproduct(String imageproduct) {
        this.imageproduct = imageproduct;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

    public String getEnlace() {
        return enlace;
    }

    public void setEnlace(String enlace) {
        this.enlace = enlace;
    }

}

File: PHP_API/DTO/DTOProduct.php
<?php

namespace App\DTO;

use JsonSerializable;

class DTOProduct implements JsonSerializable{

    public $nombre;

    public $marca;

    public $categoria;

    public $imageproduct;

    public $enlace;

    public $fecha;

    public function __construct($nombre, $marca, $categoria, $imageproduct,$fecha){
        $this->nombre = $nombre;
        $this->marca = $marca;
        $this->categoria = $categoria;
        $this->imageproduct = $imageproduct;
        $this->fecha = $fecha;
    }

    //Getters y Setters 

    public function getNombre(){return $this->nombre;}
    public function setNombre($nombre){return $this->nombre = $nombre;}

    public function getMarca(){return $this->marca;}
    public function setMarca($marca){return $this->marca = $marca;}

    public function getCategoria(){return $this->categoria;}
    public function setCategoria($categoria){return $this->categoria = $categoria;}

    public function getImageproduct(){return $this->imageproduct;}
    public function setImageproduct($imageproduct){return $this->imageproduct = $imageproduct;}

    public function getEnlace(){return $this->enlace;}
    public function setEnlace($enlace){return $this->enlace = $enlace;}

    public function getFecha(){return $this->fecha;}
    public function setFecha($fecha){$this->fecha = $fecha;}

    
    public function jsonSerialize(): array
    {
        return [
            'nombre' => $this->nombre,
            'marca' => $this->marca,
            'categoria' => $this->categoria,
            'imageproduct' => $this->imageproduct,
        ];
    }


}

?>